examples/todomvc/src/scheduler.rs

```diff
@@ -0,0 +1,138 @@
+use crate::controller::Controller;
+use crate::dbg;
+use crate::view::View;
+use crate::Message;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+/// Creates an event loop that starts each time a message is added
+pub struct Scheduler {
+    controller: Rc<RefCell<Option<Controller>>>,
+    view: Rc<RefCell<Option<View>>>,
+    events: RefCell<Vec<Message>>,
+    running: RefCell<bool>,
+}
+
+impl Scheduler {
+    /// Construct a new `Scheduler`
+    pub fn new() -> Scheduler {
+        Scheduler {
+            controller: Rc::new(RefCell::new(None)),
+            view: Rc::new(RefCell::new(None)),
+            events: RefCell::new(Vec::new()),
+            running: RefCell::new(false),
+        }
+    }
+
+    pub fn set_controller(&self, controller: Controller) {
+        if let Ok(mut controller_data) = self.controller.try_borrow_mut() {
+            controller_data.replace(controller);
+        } else {
+            dbg("This might be a deadlock");
```

Comment:
Should this be a panic instead of a debug log? (same with many of the others below)

Reply:
This was also from https://rustwasm.github.io/book/game-of-life/code-size.html#disable-debug-symbols maybe use `process::abort()` within the debug fn?

Reply:
Ah I mean that `dbg` returns after logging, but these conditions seem like they mean the application is in an erroneous state so it should somehow abort, although I'm fine either aborting via `process::abort()` or through `panic!`

Reply:
I have moved to change this to be called `exit` which triggers an exception in the JS and `process::abort` annoyingly it doesn't actually stop the JS or wasm from working though.